solver/solver.go

```diff
@@ -59,6 +62,12 @@ func Solve(doneCh chan struct{}, syncer *diff.Syncer,
 			// fire the request to Kong
 			result, err = r.Do(e.Kind, e.Op, e)
 			if err != nil {
+				// Make only 500 errors retryable - this is what Kong returns when e.g. the
+				// database is temporarily overloaded.
+				// TODO: Replace with better check when/if go-kong exposes status code
+				if !strings.HasPrefix(err.Error(), "500") {
```

Comment:
Can we make this a little more robust by checking against a regex instead to account for 501/502/503 or rather 5xx?

I've also opened up https://github.com/Kong/go-kong/issues/10 to make this better in future.

Reply:
I'm not sure if anything other than 500 can be a transient error? The trouble with retrying a permanent error would be that the decK run will take a long time while it is retrying every request the maximum number of times. So far I've only seen 500 error be transient, e.g. when the database was overloaded with too many requests. So would it not make sense to be conservative here?

Reply:
In practice, a lot of times there is another hop or proxy between decK and the Admin endpoint of Kong. That intermediate layer can send a 502 or 503 intermittently. This is certainly a hypothetical scenario and we can defer the change.

Reply:
Yeah, I see how there can be other transient errors. I like the idea of keeping the change small and restricted to just the specific case where we know decK does not do what a user would expect (load a large config with 1000s/10000s of Kong entities). We can always easily expand the reach if it comes up?